conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -576,8 +560,13 @@ private EnrichedField createField(FieldName fieldName, FieldDefinition field) {
         TypeName typeName =
                 ConjureAnnotations.withSafety(typeMapper.getClassName(type), safetyEvaluator.getUsageTimeSafety(field));
         FieldSpec.Builder spec = FieldSpec.builder(typeName, JavaNameSanitizer.sanitize(fieldName), Modifier.PRIVATE);
-        if (type.accept(TypeVisitor.IS_LIST) || type.accept(TypeVisitor.IS_SET) || type.accept(TypeVisitor.IS_MAP)) {
-            spec.initializer("new $T<>()", type.accept(COLLECTION_CONCRETE_TYPE));
+        if (type.accept(TypeVisitor.IS_LIST) || type.accept(TypeVisitor.IS_SET)) {
+            spec.initializer(
+                    "$1T.new$2L()",
```

Comment:
Calling factory methods from ConjureCollections means we don't need javapoet type($T) to be added to the import list. This causes edge cases when the collection here is the only usage. I think using an internal collection type may be easier since we can then just make the name correspond to the ConjureCollections factory method.

Reply:
I don't follow -- can you elaborate on the edge case you're describing?

Reply:
So the old initializer was `$1T.new$2T()` where $2 was passed a class, like ArrayList. The codegen adds any types added with $T to the list of deps to import. When we switched to using conjure collections factories like `ConjureCollections.newList()` this works but breaks when the name of the factory method uses a class that isn't used anywhere else and is referred to with the parent class. Ie newDoubleArrayList -> List<Double>. This would add DoubleArrayList as a import even though it isn't need. So maybe not exactly an edge case, but something that broke with the old logic.

Reply:
Ah I see, seems like the old logic was a bit incorrect/fragile :-]